### ISSUE TITLE
ArchLinux on ARM/RaspberryPi - Root-cross toolchain only,  root-native build fails

### ISSUE THREAD

**Issue by Belxjander** [sba1/adtools#16]
Cross-Compilation succeeded only once I hand-patched binutils sources,

Request change to a known stable 2.23.2 versuon archive,  possibly update to recent stable to replace
Existing sources.

I recall having to hand-patch (using latest stable as reference) several files in relation to a -1 being shifted,
Usually setting a "-(1 << CONSTANT_VALUE)" replacing the "( -1 << PPC_OPSHIFT_INV)" usage.

The root-native second pass building an AOS4 compiler itself failed as the clib2 and newlib headers are lacking a stdlib.h when building the c-lang.c file within the repo/gcc/c/ folder.

I am successfully attempting to test build with failure to link __gcov_... Symbols with the root-cross toolchain.  (see separate issue for this)

**Comment by sba1** [comment]
Please supply a patch for binutils fixes. Using a newer binutils is planned but it not sure when it is done. Contributions are always welcome.

**Comment by sba1** [comment]
> Please supply a patch for binutils fixes. Using a newer binutils is planned but it not sure when it is done. Contributions are always welcome.

I did this myself in the meantime. Here, binutils compile again with gcc 6.

**Comment by sba1** [comment]
> The root-native second pass building an AOS4 compiler itself failed as the clib2 and newlib headers are lacking a stdlib.h when building the c-lang.c file within the repo/gcc/c/ folder.

Please provide a log for this. While the native build via travis doesn't yet fully work, this this step will already be passed successfully.

**Comment by sba1** [comment]
Closing this now.
